src/App.tsx

```diff
@@ -1,14 +1,56 @@
-import React from 'react';
+import React, { useState, useMemo, useCallback } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import { Person } from './types/Person';
+import { PeopleList } from './components/DropdownList';
+
+function debounce(callback: (...args: string[]) => void, delay: number) {
+  let timeId = 0;
+
+  return (...args: string[]) => {
+    window.clearTimeout(timeId);
+
+    timeId = window.setTimeout(() => {
+      callback(...args);
+    }, delay);
+  };
+}
+
+function filterPeople(people: Person[], appliedQuery: string) {
+  return people.filter((person) => person.name
+    .toLowerCase()
+    .includes(appliedQuery.toLowerCase()));
+}
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [query, setQuery] = useState('');
+  const [appliedQuery, setAppliedQuery] = useState('');
+
+  const applyQuery = useCallback(debounce(setAppliedQuery, 1000), []);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setQuery(event.target.value);
+    applyQuery(event.target.value);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return filterPeople(peopleFromServer, appliedQuery);
+  }, [appliedQuery]);
+
+  const onSelected = useCallback((newSelectPerson: Person) => {
+    setSelectedPerson(newSelectPerson);
+    setQuery(newSelectPerson.name);
```

Comment:
you may also  immediately set `appliedQuery` to avoid appearing the previous suggestion list after selection
[Screencast from 08-10-23 12:18:49.webm](https://github.com/mate-academy/react_autocomplete/assets/86923089/408f7ae3-786b-4b31-999f-0d0c9512d1e6)